docs/ui/Commandk.tsx

```diff
@@ -0,0 +1,108 @@
+import { siteConfig } from '@/lib/config';
+import { Button, Dialog } from '@/ui';
+import { Command, CommandGroup } from 'cmdk';
+import { allContentPages } from 'contentlayer/generated';
+import React from 'react';
+
+import { useRouter } from 'next/navigation';
+
+import { Search } from '@marigold/icons';
+
+export const CommandMenu = () => {
+  // Toggle the menu when ⌘K is pressed
+  React.useEffect(() => {
+    const handleKeyDown = (e: {
+      key: string;
+      metaKey: boolean;
+      ctrlKey: boolean;
+      preventDefault: () => void;
+    }) => {
+      if (e.key === 'k' && (e.metaKey || e.ctrlKey)) {
+        e.preventDefault();
+        setOpen(open => !open);
+      }
+    };
+    document.addEventListener('keydown', handleKeyDown);
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, []);
+  const router = useRouter();
+  const [open, setOpen] = React.useState(false);
+  const [search, setSearch] = React.useState('');
+
+  const groupedPages = siteConfig.navigation.map(({ name, slug }) => {
+    const items = allContentPages
+      .filter(page => page.slug.includes(slug))
+      .map(({ title, slug, order }) => ({ title, slug, order }));
+    //sort by order or alphabetticly
+    for (let i = 0; i < items.length; i++) {
```

Comment:
There's more to it. Tip: You can write it in such a way that you don't have any nested for-loops.